wordle-app/scripts/game.ts

```diff
@@ -0,0 +1,166 @@
+  public maxAttempts: number;
+  public guesses: Word[] = [];
+  public guessIndex: number = 0;
+  public gameState: GameState = GameState.Playing;
+  public guessedLetters: Letter[] = [];
+  public isBusy: boolean = false;
+  public stats: GameStats | null = null;
+
+
+  private _secretWord: string = "";
+  private set secretWord(value: string) {
+    this._secretWord = value.toUpperCase();
+  }
+  public get secretWord(): string {
+    return this._secretWord;
+  }
+
+  constructor(maxAttempts: number = 6) {
+    this.maxAttempts = maxAttempts;
+    this.isBusy = true;
+    this.gameState = GameState.Playing;
+  }
+
+  public async startNewGame(word?: string | undefined) {
+    // Load the game
+    this.isBusy = true;
+
+    // Reset default values
+    this.guessIndex = 0;
+    this.guessedLetters = [];
+    this.stats = null;
+
+    // Get a random word
+    if(!word){
+      this.secretWord = await this.getWordOfTheDayFromApi();
+    }
+    else{
+      this.secretWord = word;
+    }
+
+    // Populate guesses with the correct number of empty words
+    this.guesses = [];
+    for (let i = 0; i < this.maxAttempts; i++) {
+      this.guesses.push(
+        new Word({ maxNumberOfLetters: this.secretWord.length })
+      );
+    }
+
+    // Start the game
+    this.gameState = GameState.Playing;
+    this.isBusy = false;
+  }
+
+  public get guess() {
+    return this.guesses[this.guessIndex];
+  }
+
+  public setGuessLetters(word: string){
+    // Loop through the word and add new letters
+    this.guess.clear();
+    for (let i = 0; i < word.length; i++) {
+      this.addLetter(word[i].toUpperCase());
+    }
+  }
+
+  public removeLastLetter() {
+    if (this.gameState === GameState.Playing) {
+      this.guess.removeLastLetter();
+    }
+  }
+
+  public addLetter(letter: string) {
+    if (this.gameState === GameState.Playing) {
+      this.guess.addLetter(letter);
+    }
+  }
+
+  public updateGuessedLetters() {
+    for (const letter of this.guess.letters) {
+      // Find the index of the letter in the guessed letters array
+      const index = this.guessedLetters.findIndex(
+        (existingLetter) => existingLetter.char === letter.char
+      );
+      if (index !== -1) {
+        // Do not update the letter if it is already correct
+        if (this.guessedLetters[index].state !== LetterState.Correct) {
+          // Do not update the letter if it is wrong
+          if (letter.state !== LetterState.Wrong) {
+            this.guessedLetters[index] = letter;
+          }
+        }
+      } else {
+        // If letter does not already exist, add it to the array
+        this.guessedLetters.push(letter);
+      }
+    }
+  }
+
+  public submitGuess() {
+    console.log("SubmitGuess")
+    if (this.gameState !== GameState.Playing) return;
+    if (!this.guess.isFilled) return;
+    if (!this.guess.isValidWord()) {
+      this.guess.clear();
+      return;
+    }
+
+    const isCorrect = this.guess.compare(this.secretWord);
```

Comment:
I wanted to submit a little bug report: When the word was "nicee", a common word that was added by a user, I was unable to submit the word as a guess. I could send "nicer", which would get all but the last letter correct, but upon trying to send "nicee", it would shake and clear the text box.
My guess is that your code for testing a word's correctness was still using the old wordlist (not pulled from the API), but I couldn't find evidence of that in your code. I may have just missed it, or it could be another cause altogether.

Reply:
Great catch. Thank you for this bug report. I totally forgot to check this. I will go and fix this now. Thanks again 👍